test/extended/images/mysql_replica.go

```diff
@@ -0,0 +1,143 @@
+	defer g.GinkgoRecover()
+	var (
+		persistentVolumes []*kapi.PersistentVolume
+		templatePath      = "https://raw.githubusercontent.com/openshift/mysql/master/5.5/examples/replica/mysql_replica.json"
+		oc                = exutil.NewCLI("mysql-replication", exutil.KubeConfigPath())
+		user              = "user"
+		userPass          = "userpass"
+		rootPass          = "rootpass"
+		database          = "userdb"
+	)
+
+	g.Describe("MySQL replication", func() {
+		g.JustBeforeEach(func() {
+			var err error
+			persistentVolumes, err = exutil.SetupHostPathVolumes(oc.AdminKubeREST().PersistentVolumes(), "mysql-replication", "512Mi", 1)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		g.AfterEach(func() {
+			err := oc.AsAdmin().Run("delete").Args("all", "--all", "-n", oc.Namespace()).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			err = oc.AsAdmin().Run("delete").Args("pvc", "--all", "-n", oc.Namespace()).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			err = exutil.CleanupHostPathVolumes(oc.AdminKubeREST().PersistentVolumes(), "mysql-replication")
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		// TODO: Add parallel loop for 5.6 templates
+		g.It("should replicate data", func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			g.By(fmt.Sprintf("processing and creating the template %q", templatePath))
+			configFile, err := oc.Run("process").
+				Args(
+				"-f", templatePath,
+				"-v", "MYSQL_USER="+user+",MYSQL_PASSWORD="+userPass+",MYSQL_ROOT_PASSWORD="+rootPass+",MYSQL_DATABASE="+database).
+				OutputToFile("config.json")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = oc.Run("create").Args("-f", configFile).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			createHelpers := func(masterDeployment, slaveDeployment string, slaveCount int) (*MySQL, []*MySQL) {
+
+				podNames, err := exutil.WaitForPods(oc.KubeREST().Pods(oc.Namespace()), exutil.ParseLabelsOrDie(fmt.Sprintf("deployment=%s", masterDeployment)), 1, 60*time.Second)
+				o.Expect(err).NotTo(o.HaveOccurred())
+				masterPod := podNames[0]
+
+				slavePods, err := exutil.WaitForPods(oc.KubeREST().Pods(oc.Namespace()), exutil.ParseLabelsOrDie(fmt.Sprintf("deployment=%s", slaveDeployment)), slaveCount, 120*time.Second)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				// Create MySQL helper for master
+				master, err := NewMysql(oc, masterPod, "")
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(master.WaitUntilUp(120 * time.Second)).NotTo(o.HaveOccurred())
+
+				// Create MySQL helpers for slaves
+				slaves := make([]*MySQL, len(slavePods))
+				for i := range slavePods {
+					slave, err := NewMysql(oc, slavePods[i], masterPod)
+					o.Expect(err).NotTo(o.HaveOccurred())
+					o.Expect(slave.WaitUntilUp(120 * time.Second)).NotTo(o.HaveOccurred())
+					slaves[i] = slave
+				}
+
+				return master, slaves
+			}
+
+			assertReplicationIsWorking := func(master *MySQL, slaves []*MySQL) {
```

Comment:
I'll move the one that creates the helpers, but keep this one more or less as it is. It is what ginkgo documentation recommends: https://onsi.github.io/ginkgo/#shared-example-patterns